test/one/profiler/test/Runner.java

```diff
@@ -210,6 +215,55 @@ private static void printSummary(EnumMap<TestStatus, Integer> statusCounts, List
         System.out.println("TOTAL: " + testCount);
     }
 
+    private static RunnerDeclaration parseRunnerDeclaration(String[] args) {
+        List<String> testNames = new ArrayList<>();
+        List<String> testDirectories = new ArrayList<>();
+        List<String> testFilters = new ArrayList<>();
+
+        // If 'arg' contains a period and not starts with a lowercase character, treat it as a testName.
+        // Otherwise, if it is an existing directory, consider it a directory.
+        // Else, treat it as a filter.
+        for (int i = 0; i < args.length; i++) {
+            String arg = args[i];
+
+            if (arg.indexOf('.') >= 0 && !Character.isLowerCase(arg.charAt(0))) {
+                try {
+                    Class.forName(arg);
+                    testNames.add(arg);
+                    continue;
+                } catch(ClassNotFoundException _e) {
+                    testFilters.add(args[i]);
+                    continue;
+                }
+            }
+
+            File f = new File("test/test/" + args[i]);
+            if (!f.exists() || !f.isDirectory()) {
+                testFilters.add(args[i]);
+            } else {
+                testDirectories.add(args[i]);
+            }
+        }
+
+        if (testDirectories.isEmpty()) {
```

Comment:
This conditional branch makes it even more confusing.
Maybe we can always get a list of all available tests and then let specifiers (or filters, whatever you call it) select matched ones?